### PR TITLE
Fix tests.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | https://github.com/yiisoft/db/issues/108

- [x] Rename namespace.
- [x] Remove code duplicate test db and db-pgsql.
- [x] Move tests db --> db-mysql.
- [x] Run tests from db-mysql.
- [x] Apply update https://github.com/yiisoft/yii2/pull/17867

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,11 +31,7 @@
     },
     "autoload": {
         "psr-4": {
-            "Yiisoft\\Db\\Mysql\\": "src"
-        }
-    },
-    "autoload-dev": {
```
Comment (xepozz) [inline composer.json]: Why did you delete this?

File: tests/ColumnSchemaBuilderTest.php [hunk 2]
```diff
@@ -2,30 +2,31 @@
...
-namespace Yiisoft\Db\Tests\Mysql;
+namespace Yiisoft\Db\Mysql\Tests;
 
 use Yiisoft\Db\Mysql\ColumnSchemaBuilder;
-use Yiisoft\Db\Schemas\Schema;
+use Yiisoft\Db\Mysql\Schema;
 use Yiisoft\Db\Tests\ColumnSchemaBuilderTest as AbstractColumnSchemaBuilderTest;
 
-final class ColumnSchemaBuilderTest extends AbstractColumnSchemaBuilderTest
+class ColumnSchemaBuilderTest extends AbstractColumnSchemaBuilderTest
```
Comment (xepozz) [inline tests/ColumnSchemaBuilderTest.php]: Why did you delete `final`?

File: tests/ConnectionTest.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+use Yiisoft\Db\Drivers\Connection;
+use Yiisoft\Db\Helper\Dsn;
+use Yiisoft\Db\Tests\ConnectionTest as AbstractConnectionTest;
+
+class ConnectionTest extends AbstractConnectionTest
+{
+    protected ?string $driverName = 'mysql';
+
+    public function testDsnHelper(): void
```
Comment (xepozz) [inline tests/ConnectionTest.php]: Can we transfer this testcase to parent class and fill in subclass only with different paramaters?

File: tests/ConnectionTest.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        $dsn = new Dsn('mysql', '127.0.0.1', 'yiitest', '3306');
+
+        $connection = new Connection($this->cache, $this->logger, $this->profiler, $dsn->getDsn());
```
Comment (xepozz) [inline tests/ConnectionTest.php]: It there is need to check the result of `getDsn()` of `Connection`? I think we should check only `Dsn` class.

File: tests/ConnectionTest.php (hunk 3 above)
Comment (terabytesoftw) [reply tests/ConnectionTest.php]: I tell you, there are currently many tests that are marked as markTestSkipped, precisely because all implementations do not support it, the idea is that each implementation has its test without skipping any.

File: tests/ConnectionTest.php (hunk 4 above)
Comment (terabytesoftw) [reply tests/ConnectionTest.php]: The test checks both.

File: tests/ColumnSchemaBuilderTest.php (hunk 2 above)
Comment (terabytesoftw) [reply tests/ColumnSchemaBuilderTest.php]: Because I am going to run the tests from each implementation in db, it currently works the other way around, and the idea is to move the tests to each implementation you can read the issue yiisoft/db#108.

File: composer.json (hunk 1 above)
Comment (terabytesoftw) [reply composer.json]: Because I am going to run the tests from each implementation in db, it currently works the other way around, and the idea is to move the tests to each implementation you can read the issue yiisoft/db#108.

File: tests/ConnectionTest.php (hunk 4 above)
Comment (xepozz) [reply tests/ConnectionTest.php]: Why? `Connection` just pass any `string|null` to `$dsn` property. We no need to check this.

Comment (terabytesoftw) [reply tests/ConnectionTest.php]: In the constructor accepts only string dsn, and for the purposes of initializing the variable I prefer null which means that it has no status, to assign a blank string.

```php
public function __construct(?CacheInterface $cache, LoggerInterface $logger, Profiler $profiler, string $dsn)
```

Comment (xepozz) [reply tests/ConnectionTest.php]: I am about that Connection is just unnecessary to test `Dsn::getDsn`
